okonomiyaki/file_formats/_package_info.py

```diff
@@ -69,7 +74,10 @@ def from_egg(cls, path_or_file):
                 raise OkonomiyakiError(msg)
         else:
             data = path_or_file.read(_PKG_INFO_LOCATION)
-            data = data.decode(PKG_INFO_ENCODING)
+            with _keep_position(path_or_file.fp):
+                sha256 = compute_sha256(_keep_position(path_or_file.fp))
```

Comment:
Do you need the `_keep_position` _context_ here as well as the call in `compute_sha256` argument?